apps/dash-stitching/app.py

```diff
@@ -0,0 +1,483 @@
+                ),
+                goButtonTitle="Estimate translation",
+            ),
+            html.Div(
+                children=[html.Button("Upload demo data")],
+                id="demo",
+                className="demo_button",
+            ),
+        ]
+    elif tab == "result-tab":
+        return [
+            dcc.Loading(
+                id="loading-1",
+                children=[
+                    html.Img(
+                        id="stitching-result",
+                        src=array_to_data_url(
+                            np.zeros((height, width), dtype=np.uint8)
+                        ),
+                        width=canvas_width,
+                    )
+                ],
+                type="circle",
+            ),
+            html.Div(
+                [
+                    html.Label("Contrast"),
+                    dcc.Slider(
+                        id="contrast-stitch", min=0, max=1, step=0.02, value=0.5
+                    ),
+                ],
+                # style={'width': '70%'},
+                className="result_slider",
+            ),
+            html.Div(
+                [
+                    html.Label("Brightness"),
+                    dcc.Slider(
+                        id="brightness-stitch", min=0, max=1, step=0.02, value=0.5
+                    ),
+                ],
+                # style={'width': '70%'},
+                className="result_slider",
+            ),
+        ]
+    else:
+        return [
+            html.Img(
+                id="bla", src=app.get_asset_url("stitch_demo.gif"), width=canvas_width
+            )
+        ]
+
+
+@app.callback(Output("stitching-tabs", "value"), [Input("button-stitch", "n_clicks")])
+def change_focus(click):
+    if click:
+        return "result-tab"
+    return "canvas-tab"
+
+
+@app.callback(Output("table-stitch", "data"), [Input("canvas-stitch", "json_data")])
+def estimate_translation(string):
+    props = parse_jsonstring_line(string)
+    if props is not None and len(props) > 0:
+        df = pd.DataFrame(props, columns=list_columns)
+        return df.to_dict("records")
+    else:
+        raise PreventUpdate
+
+
+@app.callback(Output("memory-stitch", "data"), [Input("button-stitch", "n_clicks")])
+def update_store(click):
+    sleep(0.1)
+    return click
+
+
+@app.callback(
+    Output("stitching-result", "src"),
+    [
+        Input("contrast-stitch", "value"),
+        Input("brightness-stitch", "value"),
+        Input("stitched-res", "children"),
+    ],
+)
+def modify_result(contrast, brightness, image_string):
+    if image_string is None:
+        raise PreventUpdate
+    img = np.asarray(image_string_to_PILImage(image_string))
+    img = contrast_adjust(img, contrast)
+    img = brightness_adjust(img, brightness)
+    return array_to_data_url(img)
+
+
+@app.callback(
+    Output("stitched-res", "children"),
+    [Input("button-stitch", "n_clicks")],
```

Comment:
The input for this in `modify_content` is stored in the param `n_c1` but this param is never used in the function. You can probably remove this.

Reply:
On that note, if you delete this line, you can change the following lines that are currently `States` into `Input`s.

Reply:
You still didn't change the `n_c1` param.